integrations/access/accessrequest/app.go

```diff
@@ -389,6 +467,41 @@ func (a *App) getMessageRecipients(ctx context.Context, req types.AccessRequest)
 	return recipientSet.ToSlice()
 }
 
+func (a *App) recipientsFromAccessMonitoringRules(ctx context.Context, req types.AccessRequest) *common.RecipientSet {
+	log := logger.Get(ctx)
+	recipientSet := common.NewRecipientSet()
+
+	// This switch is used to determine which plugins we are enabling access monitoring notification rules for.
+	switch a.pluginType {
+	// Enabled plugins are added to this case.
+	case types.PluginTypeSlack, types.PluginTypeOpsgenie:
+		log.Debug("Applying access monitoring rules to request")
+	default:
+		return &recipientSet
+	}
+
+	a.accessMonitoringRules.RLock()
+	defer a.accessMonitoringRules.RUnlock()
+
+	for _, rule := range a.accessMonitoringRules.rules {
+		match, err := matchAccessRequest(rule.Spec.Condition, req)
+		if err != nil {
+			log.WithError(err).Warn("Failed to parse access monitoring notification rule")
+		}
+		if match {
+			for _, recipient := range rule.Spec.Notification.Recipients {
+				rec, err := a.bot.FetchRecipient(ctx, recipient)
```

Comment:
What does this call expected todo?

Reply:
Fetch recipients converts the recipients into whatever format the plugin needs. This varies by plugin.